cloud/ceph/ceph_osd_pool.py

```diff
@@ -0,0 +1,163 @@
+version_added: "1.9"
+options:
+    name:
+        required: true
+        description:
+            - The name of the pool to create or delete.
+    state:
+        required: false
+        default: "present"
+        choices: [ "present", "absent" ]
+        description:
+            - Whether the pool should be present or not on the remote host.
+    pgnum:
+        required: false
+        description:
+            - Required for C(state=present). The total number of placement groups for the pool.
+    pgpnum:
+        required: false
+        description:
+            - Required for C(type=erasure) or when C(ruleset) is specified. The total number of placement
+              groups for placement purposes. This should be equal to the total number of placement groups,
+              except for placement group splitting scenarios. Picks up default or Ceph configuration value
+              if not specified.
+    type:
+        required: false
+        default: "replicated"
+        choices: [ "replicated", "erasure" ]
+        description:
+            - The pool type which may either be C(replicated) to recover from lost OSDs by keeping
+              multiple copies of the objects or C(erasure) to get a kind of generalized RAID5 capability.
+              The C(replicated) pools require more raw storage but implement all Ceph operations.
+              The C(erasure) pools require less raw storage but only implement a subset of the available operations.
+    ruleset:
+        required: false
+        description:
+            - The name of the crush ruleset for this pool. If specified ruleset doesn’t exist, the creation of
+              C(replicated) pool will fail with C(-ENOENT).
+
+requirements:
+ - ceph
+author: Alistair Israel
+'''
+
+EXAMPLES = '''
+# Create a pool
+- ceph_osd_pool: state=present name=data pgnum=128 pgpnum=128
+'''
+
+import datetime
+import json
+
+import rados
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state = dict(choices=['present', 'absent'], default='present'),
+            name = dict(required=True),
+            pgnum = dict(type='int'),
+            pgpnum = dict(type='int'),
+            type = dict(choices=['replicated', 'erasure'], default='replicated'),
+            ruleset = dict()
+        ),
+        supports_check_mode = False
+    )
+
+    name = module.params['name']
+
+    result = {}
+    result['name'] = name
+
+    def sh(cmd):
+        startd = datetime.datetime.now()
+        rc, out, err = module.run_command(cmd)
+        endd = datetime.datetime.now()
+        delta = endd - startd
+        result['rc'] = rc
+        result['cmd'] = cmd
+        result['stdout'] = out.rstrip("\r\n")
+        result['stderr'] = err.rstrip("\r\n")
+        result['start'] = str(startd)
+        result['end'] = str(endd)
+        result['delta'] = str(delta)
+
+    changed = False
+
+    with rados.Rados(conffile='/etc/ceph/ceph.conf') as cluster:
```

Comment:
What about multi-cluster support?
Shouldn't we expose a ceph_config variable that will take `/etc/ceph/ceph.conf` for default so the user can override it if necessary?